src/plugin.js

```diff
@@ -199,7 +204,7 @@ function doZoom(chart, percentZoomX, percentZoomY, focalPoint, whichAxes) {
 
 		// Which axe should be modified when figers were used.
 		var _whichAxes;
-		if (zoomMode === 'xy' && whichAxes !== undefined) {
+		if (directionEnabled(chart, zoomMode, 'x') && directionEnabled(chart, zoomMode, 'y') && whichAxes !== undefined) {
```

Comment:
Rather than pass `chart` into `directionEnabled`, I would resolve the option in `resolveOptions`. The reason I suggest that is that it'd be a bit more consistent with how we handle options in the main Chart.js library. You could perhaps do something like:

```
if (typeof options.zoom.mode === 'function') {
  options.zoom.mode = options.zoom.mode({chart: chart});
}
props._options = options;
```